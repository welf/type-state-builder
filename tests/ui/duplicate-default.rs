use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct DuplicateDefault {
    name: String,

    // This should be an error: duplicate default attributes
    #[builder(default = "42")]
    #[builder(default = "100")]
    count: u32,
}

fn main() {}
