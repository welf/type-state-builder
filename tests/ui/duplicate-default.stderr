error: Duplicate default attribute. Only one default is allowed per field
 --> tests/ui/duplicate-default.rs:9:15
  |
9 |     #[builder(default = "100")]
  |               ^^^^^^^^^^^^^^^
